### PR TITLE
Replace local Cromwell Docker with a deployed API

### PR DESCRIPTION
**Pull request by mckinsel**
This change replaces the previous version of the local-backend Cromwell
test environment. It used to be a docker container but now is an API
deployed via AWS API Gateway.

### CODE REVIEW

File: external_executors/local_cromwell/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+This deploys an API on AWS that looks a little bit like WES and runs WDLs using
+Cromwell with a local backend. Its purpose is to provide one portability test
+environment for the HCA's secondary analysis workflows.
+
+There are two files:
+
+### local_cromwell.py
+This defines two Lambda functions that are responsible for handling two
+request types: POSTs to `/workflows` and GETs to
+`/worfkflows/{workflow_id}/status`. These two calls are enough to launch a test
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/README.md]: Minor typo in `worfkflows`

File: external_executors/local_cromwell/local_cromwell.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+fi
+
+trap - EXIT
+shutdown -H 5
+"""
+
+# Define boto3 clients
+EC2 = boto3.client('ec2', region_name=REGION)
+
+def workflows_post(event, context):
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell.py]: Might be helpful to note the python type of event (dict?) and context in the docstring.

File: external_executors/local_cromwell/local_cromwell.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        MinCount=1,
+        MaxCount=1,
+        InstanceInitiatedShutdownBehavior='terminate',
+        UserData=run_script,
+        IamInstanceProfile={
+            "Arn": ec2_profile
+        }
+    )
+
+    #instance_id = instance['Instances'][0]['InstanceId']
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell.py]: Is this residual or meant to inform how to get the instance ID? If the latter, maybe be more verbose in your comment.

File: external_executors/local_cromwell/local_cromwell.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+    return {
+        "statusCode": 200,
+        "headers": {"Content-Type": "application/json"},
+        "body": '{"workflow_id": "' + job_id + '"}'
+        }
+
+def workflow_status(event, context):
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell.py]: Give the python types for event, context.

File: external_executors/local_cromwell/local_cromwell.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+    workflow_id = event["workflow_id"]
+    db_table_name = os.environ["DB_TABLE"]
+
+    ddb = boto3.resource("dynamodb", region_name=REGION)
+    table = ddb.Table(db_table_name)
+    response = table.query(
+        KeyConditionExpression=boto3.dynamodb.conditions.Key('JobId').eq(workflow_id)
+    )
+
+    if not response["Count"]:
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell.py:352]: What do you expect Count to return to trigger the 404 state?

File: external_executors/local_cromwell/local_cromwell.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+            "errorType": "NotFound",
+            "httpStatus": "404",
+            "requestId": context.aws_request_id,
+            "message": "Workflow {} was not found".format(workflow_id)
+            }
+
+        raise Exception(json.dumps(error_dict))
+
+    job_status = "Unknown"
+    for item in response["Items"]:
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell.py]: It seems like you only ever expect one item here (based on the `break` flow controls you're using). 

If that's the case, you could code this in a way that explicitly indicates that by simply querying `response['Items'][0]['EventType']`. It's ugly, but I think clearer.

File: external_executors/local_cromwell/local_cromwell_template.json [hunk 7]
```diff
@@ -0,0 +1,418 @@
+{
```
Comment (ambrosejcarr) [inline external_executors/local_cromwell/local_cromwell_template.json:1]: I haven't done enough coding of this type to provide a good criticism of this document; you might want to get another set of eyes here.

**Review by ambrosejcarr — APPROVED** [review]
Marcus, 

This was really fun to read. I left some minor comments on the local_cromwell.py file. It looks good to me, and the documentation you added was excellent. 

I'm not sure I have enough expertise to look over the local_cromwell_template.json, so you might want to get more eyes on that.

File: external_executors/local_cromwell/README.md (hunk 1 above)
Comment (mckinsel) [reply external_executors/local_cromwell/README.md]: Thanks, will fix

File: external_executors/local_cromwell/local_cromwell.py (hunk 2 above)
Comment (mckinsel) [reply external_executors/local_cromwell/local_cromwell.py]: Okay, I'll expand the docstring a little.

File: external_executors/local_cromwell/local_cromwell.py (hunk 3 above)
Comment (mckinsel) [reply external_executors/local_cromwell/local_cromwell.py]: This is an old debugging line. I'll remove it.

File: external_executors/local_cromwell/local_cromwell.py (hunk 5 above)
Comment (mckinsel) [reply external_executors/local_cromwell/local_cromwell.py:352]: when Count is 0

File: external_executors/local_cromwell/local_cromwell.py (hunk 6 above)
Comment (mckinsel) [reply external_executors/local_cromwell/local_cromwell.py]: I'm going to break this out into a more clearly explained function. But there are multiple of events that should be recorded, and they'll be returned in an arbitrary order (I think). The logic is that there are two terminal states, Failed and Succeeded, and once we see those we know the state of the job and can stop looking. However there are intermediate states too. So in this case if we see a Started but not a Failed or Succeeded, we know the job is running but not done. If we don't even see a Started, then we don't know what's going on yet.

**Pull request merged**